Integrations/Snowflake/Snowflake.py

```diff
@@ -0,0 +1,392 @@
+    incident['name'] = name
+    # Incident occurrence time as timestamp - the datetime field specified in the integration parameters
+    incident['timestamp'] = timestamp
+    # The raw response for the row (reformatted to be json serializable) returned by the db query
+    reformatted_row = format_to_json_serializable(column_descriptions, row)
+    incident['rawJSON'] = json.dumps(reformatted_row)
+    return incident
+
+
+'''MAIN FUNCTIONS / API CALLS'''
+
+
+def test_module():
+    """
+    Test the validity of the integration instance parameters by trying to create a connection
+
+    returns:
+        An 'ok' message if valid, otherwise an error message
+    """
+    conn = get_connection({})
+    demisto.results('ok')
+    conn.close()
+
+
+def fetch_incidents():
+    """
+    Fetch events from this integration and return them as Demisto incidents
+
+    returns:
+        Demisto incidents
+    """
+    # demisto.getLastRun() will returns an obj with the previous run in it.
+    last_run = demisto.getLastRun()
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetched_data_timestamp')
+    first_fetch = False
+
+    # Handle first time fetch, fetch incidents retroactively
+    if not last_fetch:
+        first_fetch = True
+        last_fetch, _ = parse_date_range(FETCH_TIME, to_timestamp=True)
+    args = {'rows': MAX_ROWS, 'query': FETCH_QUERY}
+
+    column_descriptions, data = snowflake_query(args)
+    # convert the data/events to demisto incidents
+    incidents = []
+    for row in data:
+        incident = row_to_incident(column_descriptions, row)
+        incident_timestamp = incident.get('timestamp')
+
+        # Update last run and add incident if the incident is newer than last fetch
+        if incident_timestamp and incident_timestamp >= last_fetch:
+            if first_fetch:
+                last_fetch = incident_timestamp
+                incidents.append(incident)
+                first_fetch = False
+            elif len(incidents) >= 1 and incident.get('rawJSON') != incidents[-1].get('rawJSON'):
+                last_fetch = incident_timestamp
+                incidents.append(incident)
+
+    demisto.setLastRun({'last_fetched_data_timestamp': last_fetch})
+    demisto.incidents(incidents)
+
+
+def snowflake_query(args):
+    connection = get_connection(args)
+    if connection:
+        try:
+            query = args.get('query')
+            cur = connection.cursor(snowflake.connector.DictCursor)
+            cur.execute(query)
+            rows = args.get('rows')
+            if rows:
+                rows = int(rows)
+            if rows and rows > MAX_ROWS:
+                rows = MAX_ROWS
+            results = cur.fetchmany(rows)
+            if results:
+                return cur.description, results
+            else:
+                return_error('No data found matching the query')
```

Comment:
why is this an error?

Reply:
Should it be a regular war room message instead?